components/camel-spring-rabbitmq/src/main/java/org/apache/camel/component/springrabbit/DefaultMessagePropertiesConverter.java

```diff
@@ -39,12 +41,68 @@ public DefaultMessagePropertiesConverter(CamelContext camelContext, HeaderFilter
     @Override
     public MessageProperties toMessageProperties(Exchange exchange) {
         MessageProperties answer = new MessageProperties();
-        String contentType = ExchangeHelper.getContentType(exchange);
+        Message message = exchange.getMessage();
+
+        Object deliveryMode = message.getHeader(SpringRabbitMQConstants.DELIVERY_MODE);
+        if (deliveryMode != null) {
+            answer.setDeliveryMode(deliveryMode instanceof MessageDeliveryMode ? (MessageDeliveryMode) deliveryMode
+                    : MessageDeliveryMode.fromInt(Integer.parseInt(deliveryMode.toString())));
+        }
+        Object type = message.getHeader(SpringRabbitMQConstants.TYPE);
+        if (type != null) {
+            answer.setType(type.toString());
```

Comment:
Same here, why don't you use the method `getHeader(String name, Class<T> type)` to trigger the type conversion if needed?

Reply:
done everywhere